[Finishes #172410184] Caldera independiente, ajustes de fondos y sprites nuevos.

Adrian-de-Lucas-Gomez opened the pull request:
1º El fondo de la sala de calderas ha perdido la caldera. Se han remodelado un poco las tuberías para que tengan sentido sin importar donde esté la caldera.

2º La caldera se ha dejado como una imagen independiente para así ponerla el tamaño que queramos, en la posición que nos convenga a través del tilemap. También para poder hacer animaciones chulas :D.

3º Se ha retocado los colores de la sala de estar para adaptarlos a la paleta que se eligió ya que este fondo se creo antes de que tuviéramos una paleta de color definida. Ahora al ser colores mas flojos se puede distinguir mejor al jugador.

4º He hecho varios de los sprites de algunas de las entidades que aun tenían el place-holder, los sprites hechos son:
-Control remoto de Pelea por el Mando.
-Router del modo Wi-Figth.
-Tayunos aka monedicas del modo Capitalismo (Version de Gameplay y de UI).
-Roombas escupemonedas del modo Capitalismo con animación de cepillos. Se ha hecho un componente AnimationLoop que sirve para entidades que deben estar siempre haciendo una animación, es ajustable en velocidad u número de frames.
-Panel de control de la cinta de correr.

Adrian-de-Lucas-Gomez committed:
Sprites cambiados para mapa y caldera

Se ha tenido que cambiear toda la parte posterior de la caldera al ser ahora un objeto independiente. Se ha decidido continuar las tuberias para que no se vea raro aun cambiando el tamaño de la caldera mucho.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172410184-CalderaIndependiente

Adrian-de-Lucas-Gomez committed:
Actualizacion con el master

Adrian-de-Lucas-Gomez committed:
Update Resources.cpp

Adrian-de-Lucas-Gomez committed:
Panel de la cinta de correr

Tiene lucecitas y tres modos de sufrimiento incorporado. Al menos viene con posavasos.

Adrian-de-Lucas-Gomez committed:
Roomba rumboso

Esta añadido el spritesheet pero aun falta un componente para animar

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172410184-CalderaIndependiente

Adrian-de-Lucas-Gomez committed:
Mas texturas añadidas

Listas para verse cuando se arreglen los bugs de los objetos especiales

Adrian-de-Lucas-Gomez committed:
Animacion y constantes para Roomba

Ya se mueve haciendo la animacion de los cepillos. Se ha creado un nuevo componente llamado AnimationLoop que hace la animacion de una spritesheet horizontal. Te cambia en frame cada X milisegundos y le tiene sque dar ademas el tamaño del frame y el número de ellos.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172410184-CalderaIndependiente

Adrian-de-Lucas-Gomez committed:
Fondo adaptado a la paleta

Se ha mejorado el fondo para que sea mas ficicil de confundir con el frente de la escena. Tambien he hecho las monedas ligeramente mas grandes.

Adrian-de-Lucas-Gomez committed:
Merge branch 'master' into 172410184-CalderaIndependiente

EstebenR reviewed (changes requested):
La clase AnimationLoop mejor que herede de Viewer y se llame AnimatedViewer;
No deberia ir por tiempo si no por frames, cada vez que se llame a update pasa un frame y ya le pones tu un frameRate (cada x frames de juego se pasa al siguiente frame de animacion)

Al heredar de Viewer no tendría ni que llevar una referencia al viewer ni redefinir cualquier cosa que se pueda reutilizar

Adrian-de-Lucas-Gomez committed:
AnimatedViewer en bonito

Ahora en vez de ser un componente suento es un nuevo tipo de Viewer. Gracias a esto ahora no es necesario pasarle el numero de frames ya que lo calcula dentro. Tambien se ha abreviado un poco el codigo en el update y se ha comentado.

EstebenR committed:
Claridad de variables

mmur07 committed:
Merge branch 'master' into 172410184-CalderaIndependiente

mmur07 committed:
Revert "Merge branch 'master' into 172410184-CalderaIndependiente"

This reverts commit 0b27ca0af9a3b7f0999860ed97f1c97a8c66058c, reversing
changes made to f053d1228436495f9e4a9c5bfdfd9bf393255218.

EstebenR committed:
Revert "Revert "Merge branch 'master' into 172410184-CalderaIndependiente""

This reverts commit b483da2d254cebf4871447366c877b0fbc6b7b4d.

eljorgemo reviewed (approved):
Nice. Molaría que los roombas se orientasen a donde están yendo
